sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogUtils.scala

```diff
@@ -280,7 +280,12 @@ object CatalogUtils {
    * @return the URI of the path
    */
   def stringToURI(str: String): URI = {
-    new Path(str).toUri
+    try {
+      new Path(str).toUri
```

Comment:
can we do an empty/null check before this line?

Reply:
PTAL